de.metas.vertical.creditscore/base/src/main/java/de/metas/vertical/creditscore/base/spi/repository/TransactionResultRepository.java

```diff
@@ -0,0 +1,114 @@
+import de.metas.vertical.creditscore.base.spi.model.TransactionResult;
+import lombok.NonNull;
+import org.adempiere.ad.dao.IQueryBL;
+import org.springframework.stereotype.Repository;
+
+import java.sql.Timestamp;
+import java.util.Optional;
+
+import static org.adempiere.model.InterfaceWrapperHelper.newInstance;
+import static org.adempiere.model.InterfaceWrapperHelper.save;
+
+@Repository
+public class TransactionResultRepository
+{
+
+	public TransactionResult createTransactionResult(@NonNull final CreditScore creditScore, @NonNull final BPartnerId bPartnerId,
+			final OrderId orderId)
+	{
+		final I_CS_Transaction_Result transactionResult = newInstance(I_CS_Transaction_Result.class);
+
+		final CreditScoreRequestLogData logData = creditScore.getRequestLogData();
+		transactionResult.setRequestStartTime(Timestamp.valueOf(logData.getRequestTime()));
+		transactionResult.setRequestEndTime(Timestamp.valueOf(logData.getResponseTime()));
+		transactionResult.setResponseCode(creditScore.getResultCode().name());
+		if (creditScore.getResultCodeOverride() != null)
+		{
+			transactionResult.setResponseCodeOverride(creditScore.getResultCodeOverride().name());
+			transactionResult.setResponseCodeEffective(creditScore.getResultCodeOverride().name());
+		}
+		else
+		{
+			transactionResult.setResponseCodeEffective(creditScore.getResultCode().name());
+		}
+		transactionResult.setPaymentRule(creditScore.getPaymentRule());
+		transactionResult.setRequestPrice(creditScore.getRequestPrice());
+		if (creditScore.getCurrency() != null)
+		{
+			transactionResult.setC_Currency_ID(creditScore.getCurrency().getRepoId());
+		}
+		transactionResult.setResponseCodeText(creditScore.getResultText());
+		transactionResult.setResponseDetails(creditScore.getResultDetails());
+		transactionResult.setTransactionCustomerId(logData.getCustomerTransactionID());
+		transactionResult.setTransactionIdAPI(logData.getTransactionID());
+		transactionResult.setC_BPartner_ID(bPartnerId.getRepoId());
+		if (orderId != null)
+		{
+			transactionResult.setC_Order_ID(orderId.getRepoId());
+		}
+		save(transactionResult);
+		return mapTransactionResult(transactionResult);
+	}
+
+	public Optional<TransactionResult> getLastTransactionResult(@NonNull final String paymentRule, @NonNull final BPartnerId bPartnerId)
+	{
+		final IQueryBL queryBL = Services.get(IQueryBL.class);
+
+		final I_CS_Transaction_Result transactionResult = queryBL
+				.createQueryBuilder(I_CS_Transaction_Result.class)
+				.addOnlyActiveRecordsFilter()
+				.orderByDescending(I_CS_Transaction_Result.COLUMNNAME_RequestStartTime)
+				.addEqualsFilter(I_CS_Transaction_Result.COLUMNNAME_PaymentRule, paymentRule)
+				.addEqualsFilter(I_CS_Transaction_Result.COLUMNNAME_C_BPartner_ID, bPartnerId.getRepoId())
+				.create().first();
+		TransactionResult mappedTransactionResult = null;
+		if (transactionResult != null)
+		{
+			mappedTransactionResult = mapTransactionResult(transactionResult);
```

Comment:
I think it's important to keep a certain consistency across the project. Maybe we can reevaluate later if this will happen repeatedly with different people.